### PR TITLE
Moved audioio.WaveFile -> audiocore.WaveFile

### LINKED ISSUE #1342: audioio.WaveFile -> audiocore.WaveFile
Some guides still use WaveFile in audioio. In 5 it was moved to audiocore and in 6 it was removed from audio.

So, guides should be switched to using audiocore. It'll work on 5.x and 6.x already.

Here is a grep (a few are correct):

```
ag -Q WaveFile Adafruit_Learning_System_Guides/
Adafruit_Learning_System_Guides/HalloWing_Tour_Guide/code.py
53:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/CPB_ANCS/code.py
17:from audiocore import WaveFile
49:wave = WaveFile(file)

Adafruit_Learning_System_Guides/Burning_Fire_Wizard_Staff/code.py
92:        wave = audiocore.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Hallowing_Lightsaber/lightsaber_unicorn.py
119:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Hallowing_Lightsaber/lightsaber_standard.py
54:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/PropMaker_KeyBlade/code.py
73:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/CPX_Sound_Box/code.py
40:    with audioio.WaveFile(wave_file) as wave:

Adafruit_Learning_System_Guides/ABC_Soundboards_for_NeoTrellis/code.py
79:        wave = audioio.WaveFile(f)
102:    wav = audioio.WaveFile(f)
120:            wav = audioio.WaveFile(f)
165:            wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Grand_Central_Soundboard/code.py
37:    wav = audioio.WaveFile(audio_file)

Adafruit_Learning_System_Guides/HAL-9000/code.py
30:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/NeoTrellis_M4_Grains_of_Sand/code.py
80:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/CLUE_Rock_Paper_Scissors/advanced/rps_audio.py
23:from audiocore import WaveFile
40:    _file_buf = None  # Use for WaveFile objects
56:                fhs[file] = WaveFile(wav_file, self._file_buf)

Adafruit_Learning_System_Guides/Animatronic_Hand/binary_counting.py
25:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Christmas_Soundboard/code.py
59:        wave = audioio.WaveFile(f)
84:    wav = audioio.WaveFile(f)
105:            wav = audioio.WaveFile(f)
150:            wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Disc_Step_Sequencer/Disc_Step_Sequencer.py
30:    wav = audioio.WaveFile(f)
57:    sample = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Introducing_CircuitPlaygroundExpress/CircuitPlaygroundExpress_808_Drum_Machine.py
8:    from audiocore import WaveFile
10:    from audioio import WaveFile
46:    wave = WaveFile(file)

Adafruit_Learning_System_Guides/Introducing_CircuitPlaygroundExpress/CircuitPlaygroundExpress_AudioFiles.py
5:    from audiocore import WaveFile
7:    from audioio import WaveFile
38:    with WaveFile(wave_file) as wave:

Adafruit_Learning_System_Guides/Adafruit_UDA1334A/CircuitPython_I2S_Wave.py
6:wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/NY_Ball_Drop/code.py
227:    wav = audioio.WaveFile(sound_file)

Adafruit_Learning_System_Guides/NeoTrellis_Soundbox/code.py
72:    wav = audioio.WaveFile(audio_file)

Adafruit_Learning_System_Guides/CircuitPython_Essentials/CircuitPython_Audio_Out_Wave.py
6:    from audiocore import WaveFile
8:    from audioio import WaveFile
22:wave = WaveFile(wave_file)

Adafruit_Learning_System_Guides/HalloWing_Cat_Toy/code.py
62:    wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Mini_Chair_Swing_Ride/code.py
42:    wav = audioio.WaveFile(audio_file)

Adafruit_Learning_System_Guides/Glowing_Bottle_Castle/code.py
18:from audiocore import WaveFile
105:    WaveFile(open("sounds/F2.wav", "rb")),  # 0
106:    WaveFile(open("sounds/G2.wav", "rb")),  # 1
107:    WaveFile(open("sounds/A2.wav", "rb")),  # 2
108:    WaveFile(open("sounds/Bb2.wav", "rb")),  # 3
109:    WaveFile(open("sounds/C2.wav", "rb")),  # 4
110:    WaveFile(open("sounds/D3.wav", "rb")),  # 5
111:    WaveFile(open("sounds/E3.wav", "rb")),  # 6
112:    WaveFile(open("sounds/F3.wav", "rb")),  # 7
113:    WaveFile(open("sounds/F1.wav", "rb")),  # 7
114:    WaveFile(open("sounds/G1.wav", "rb")),  # 8
115:    WaveFile(open("sounds/A1.wav", "rb")),  # 9
116:    WaveFile(open("sounds/Bb1.wav", "rb")),  # 10
117:    WaveFile(open("sounds/C1.wav", "rb")),  # 11
118:    WaveFile(open("sounds/D2.wav", "rb")),  # 12
119:    WaveFile(open("sounds/E2.wav", "rb")),  # 13
120:    WaveFile(open("sounds/F2.wav", "rb")),  # 13
125:    WaveFile(open("sounds/undersea.wav", "rb")),
126:    WaveFile(open("sounds/tequila.wav", "rb")),
127:    WaveFile(open("sounds/lion.wav", "rb")),

Adafruit_Learning_System_Guides/LEGO_Head_CPX_Lamp/code.py
28:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Cat_Treat_Dispenser/code.py
13:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Owl_Robot/main.py
36:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/NeoTrellis_M4_RPG_Sound_Board/code.py
50:    wav = audioio.WaveFile(f)
81:            wav = audioio.WaveFile(f)
118:            wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Milk_Jug_Glow_Skull/milk_jug_glow_skull.py
51:    wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/NeoTrellis_M4_MIDI_Synth/synth.py
79:        wav = audioio.WaveFile(sound_file)
108:            wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Propmaker_Master_Sword/code.py
79:        wave = audiocore.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Crickits/ghost_slime/code.py
42:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/dont_fear_the_crickit/code.py
63:        wavf = audioio.WaveFile(file)
99:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/cpx-1701/code.py
32:wav = audioio.WaveFile(f)
55:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/climbing_raccoon/code.py
59:        wavf = audioio.WaveFile(file)

Adafruit_Learning_System_Guides/Crickits/feynman_bot/code.py
39:    wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/dannybot/code.py
52:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/yayayay_flier/code.py
46:    wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/robby_gyrobot/code.py
70:    wav = audioio.WaveFile(f)
81:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/carousel/code.py
38:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickits/yanny_or_laurel/code.py
57:wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Make_It_Talk/say-temp.py
37:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Crickit_Hello_World/code.py
13:    wav = audioio.WaveFile(audio_file)

Adafruit_Learning_System_Guides/Screaming_Painting/screaming_painting.py
43:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Hallowing_Jump_Sound/jump-sound.py
23:    return audioio.WaveFile(open(name + '.wav', 'rb'))

Adafruit_Learning_System_Guides/Hallowing_Jump_Sound/stomp-and-roar.py
26:    return audioio.WaveFile(open(name + '.wav', 'rb'))

Adafruit_Learning_System_Guides/CircuitPython_Star_Gazer/code.py
72:    wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Adafruit_MAX98357/CircuitPython_I2S_Wave.py
6:wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/NeoTrellis_M4_Simple_Drum_Machine/code.py
59:        wave = audioio.WaveFile(f)
82:    wav = audioio.WaveFile(f)
110:    sample = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/CircuitPython_WeatherCloud/code.py
19:wave = audioio.WaveFile(wave_file)
133:                wave = audioio.WaveFile(wave_file)
138:                wave = audioio.WaveFile(wave_file)
143:                wave = audioio.WaveFile(wave_file)
155:                wave = audioio.WaveFile(wave_file)
207:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Bluetooth_Restroom_Keys/code.py
8:from audiocore import RawSample, WaveFile
58:        wave = WaveFile(file)

Adafruit_Learning_System_Guides/Tightrope_Unicycle_Bot/tightrope_unicycle_bot.py
26:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/PropMaker_Light_Up_Prop/code.py
71:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Circuit_Playground_Express_Rocket_Lamp/code.py
52:    with audioio.WaveFile(wave_file) as wave:

Adafruit_Learning_System_Guides/Circuit_Playground_Bluefruit_Pumpkin/code.py
14:from audiocore import WaveFile
58:        wave = WaveFile(wave_file)

Adafruit_Learning_System_Guides/Crickit_Sparky_Automaton/code.py
34:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Slideshows_Soundtrack/slideshows_soundtrack.py
46:        wav = audioio.WaveFile(data)

Adafruit_Learning_System_Guides/PropMaker_Lightsaber/code.py
75:        wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Minesweep/code.py
22:    from audioio import WaveFile
24:    from audiocore import WaveFile
227:    wavedata = WaveFile(wavfile)

Adafruit_Learning_System_Guides/Adafruit_Prop_Maker_FeatherWing/Prop_Maker_Audio_Simpletest.py
15:    wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/Motion_Gift_Box/code.py
34:    wav = audioio.WaveFile(audio_file)

Adafruit_Learning_System_Guides/Hand_Turkey/code.py
41:        wav = audioio.WaveFile(f)

Adafruit_Learning_System_Guides/Music_Box_with_Crickit/code.py
14:audio = audioio.WaveFile(open(AUDIO_FILENAME, "rb"))

Adafruit_Learning_System_Guides/HalloWing_Jump_Scare_Trap/hallowing_jump_scare_trap.py
40:    wave = audioio.WaveFile(wave_file)

Adafruit_Learning_System_Guides/CP101_StateMachines/brute-force/code.py
230:    wav = audioio.WaveFile(sound_file)

Adafruit_Learning_System_Guides/CP101_StateMachines/classes/code.py
194:    wav = audioio.WaveFile(sound_file)

Adafruit_Learning_System_Guides/NeoTrellis_M4_Sound_Board/code.py
59:        wave = audioio.WaveFile(f)
84:    wav = audioio.WaveFile(f)
105:            wav = audioio.WaveFile(f)
150:            wav = audioio.WaveFile(f)
```

### PR DESCRIPTION
**Pull request by dherrada**
Also moved audioio.RawSample -> audiocore.RawSample

Fixes #1342

### CODE REVIEW

**Review by ladyada — APPROVED** [review]
scanned thru, didnt test

**Comment by dherrada** [comment]
I haven't ever done a whole lot with audio, so there were a lot of projects that I couldn't test since I didn't have the hardware, but here's what I did test:

- Circuit_Playground_Express_Rocket_Lamp
- Make_It_Talk (wasn't initially working, although I don't think this had anything to do with the audioio to audiocore switch)

If you'd like me to test more, I can see about grabbing a crickit and a propmaker featherwing since a majority of the projects I modified use one of those.

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
im ok with that, @tannewt any thoughts?

**Comment by dhalbert** [comment]
I was going to submit exactly the same fix for a single one of these, then discovered this PR. Very happy this is being done.

File: Make_It_Talk/say-temp.py [hunk 1]
```diff
@@ -34,15 +40,15 @@ def play_file(wavfile):
...
+        wav = audiocore.WaveFile(f)
         aout.play(wav)
         while aout.playing:
             pass
 
 # Function should take an integer -299 to 299 and say it
 # Assumes wav files are available for the numbers
 def read_temp(temp):
-    play_file("temperature.wav")
+    play_file("The temperature is.wav")
```
Comment (tannewt) [inline Make_It_Talk/say-temp.py:51]: When did the filename change? Filenames with spaces can be awkward.

**Review by tannewt — COMMENTED** [review]
One question. Everything else looks good.

File: Make_It_Talk/say-temp.py (hunk 1 above)
Comment (dherrada) [reply Make_It_Talk/say-temp.py:51]: If you look in the Make_It_Talk/numbers directory, temperature.wav doesn't exist, which causes the code to fail, and I'm like 99% certain that it was intended to be "The temperature is.wav" instead. Alternatively, I could rename it to temperature.wav, but I have tested it and it does work.

Comment (tannewt) [reply Make_It_Talk/say-temp.py:51]: Ok, this is fine. Thanks!

**Review by tannewt — APPROVED** [review]
Needs a merge

**Pull request merged**